Data/samples/DBLogger/src/DBLogger.cpp

```diff
@@ -0,0 +1,337 @@
+
+		options.addOption(
+			Option("help", "h", "display help information on command line arguments")
+				.required(false)
+				.repeatable(false)
+				.callback(OptionCallback<DBLogger>(this, &DBLogger::handleHelp)));
+
+		options.addOption(
+			Option("?", "?", "display help information on command line arguments")
+				.required(false)
+				.repeatable(false)
+				.callback(OptionCallback<DBLogger>(this, &DBLogger::handleHelp)));
+
+		options.addOption(
+			Option("dir", "d", "directory path to scan for SQL log files")
+				.required(true)
+				.repeatable(false)
+				.argument("dir")
+				.callback(OptionCallback<DBLogger>(this, &DBLogger::handleLogDirectory)));
+
+		options.addOption(
+			Option("connector", "c", "database connector")
+				.required(true)
+				.repeatable(false)
+				.argument("connector")
+				.callback(OptionCallback<DBLogger>(this, &DBLogger::handleDbConnector)));
+
+		options.addOption(
+			Option("constring", "s", "database connection string")
+				.required(true)
+				.repeatable(false)
+				.argument("cstr")
+				.callback(OptionCallback<DBLogger>(this, &DBLogger::handleDbConnectionString)));
+
+	}
+
+	void handleHelp(const std::string& name, const std::string& value)
+	{
+		_helpRequested = true;
+		displayHelp();
+		stopOptionsProcessing();
+	}
+
+	void handleLogDirectory(const std::string& name, const std::string& value)
+	{
+		if (value.empty())
+			throw Poco::Util::IncompatibleOptionsException("Expected directory name");
+		_directory = value;
+	}
+
+	void handleDbConnector(const std::string& name, const std::string& value)
+	{
+		if (value.empty())
+			throw Poco::Util::IncompatibleOptionsException("Expected database connector name");
+		_connector = value;
+	}
+
+	void handleDbConnectionString(const std::string& name, const std::string& value)
+	{
+		if (value.empty())
+			throw Poco::Util::IncompatibleOptionsException("Expected database connection string");
+		_connectionString = value;
+	}
+
+	void displayHelp()
+	{
+		HelpFormatter helpFormatter(options());
+		helpFormatter.setCommand(commandName());
+		helpFormatter.setUsage("OPTIONS");
+		helpFormatter.format(std::cout);
+	}
+
+	void onItemAdded(const Poco::DirectoryWatcher::DirectoryEvent& ev)
+	{
+		logger().trace("Added: %s", ev.item.path());
+		processFile(ev.item);
+	}
+
+	void onItemModified(const Poco::DirectoryWatcher::DirectoryEvent& ev)
+	{
+		logger().information("Modified: %s", ev.item.path());
+		processFile(ev.item);
+	}
+
+	void onItemMovedFrom(const Poco::DirectoryWatcher::DirectoryEvent& ev)
+	{
+		logger().trace("Moved from: %s", ev.item.path());
+		processFile(ev.item);
+	}
+
+	void onItemMovedTo(const Poco::DirectoryWatcher::DirectoryEvent& ev)
+	{
+		logger().trace("Moved to: %s", ev.item.path());
+	}
+
+	void onItemRemoved(const Poco::DirectoryWatcher::DirectoryEvent& ev)
+	{
+		logger().trace("Removed: %s", ev.item.path());
+	}
+
+	void processFile(const Poco::File& file)
+	{
+		// TODO: What to do with symbolic links?
```

Comment:
we don't want to bother with links to files; I think directory link should not be a problem